### PR TITLE
fix script_name without / at beginning

### PR DESCRIPTION
**Pull request by abtinmo**
**What kind of change does this PR introduce?**

  - [x] bug fix
  - [ ] feature
  - [ ] docs update
  - [ ] tests/coverage improvement
  - [ ] refactoring
  - [ ] other

**What is the related issue number (starting with `#`)**

#1772 

**What is the current behavior?** (You can also link to an open issue here)

#1772 

**What is the new behavior (if this is a feature change)?**

1.  shows an error, telling: script_name foo should start with /

2. fixes the script_name to prevent 404 error (not sure if it's necessary)

**Other information**:

**Checklist**:

  - [x] I think the code is well written
  - [x] I wrote [good commit messages][1]
  - [ ] I have [squashed related commits together][2] after the changes have been approved
  - [x] Unit tests for the changes exist
  - [x] Integration tests for the changes exist (if applicable)
  - [x] I used the same coding conventions as the rest of the project
  - [x] The new code doesn't generate linter offenses
  - [ ] Documentation reflects the changes
  - [x] The PR relates to *only* one subject with a clear title
  and description in grammatically correct, complete sentences

[1]: http://chris.beams.io/posts/git-commit/
[2]: https://github.com/todotxt/todo.txt-android/wiki/Squash-All-Commits-Related-to-a-Single-Issue-into-a-Single-Commit

### CODE REVIEW

**Comment by webknjaz** [comment]
Hey @abtinmo, thanks for the PR!

Please **add a test for this change**. And make sure the CI is green.

**Comment by abtinmo** [comment]
Hi @webknjaz,

it seems there is a problem with your macos config for ci.
and failed with  AppVeyor is for #1750

**Comment by webknjaz** [comment]
@abtinmo don't worry Circle CI and that flaky test. Just add a test case for your change.

**Comment by abtinmo** [comment]
@webknjaz its done.

File: cherrypy/test/test_core.py
Comment (webknjaz) [inline cherrypy/test/test_core.py:839]: Please also test `script_name=None`, `script_name=''` and `script_name='/'`.

It looks like it changes the behavior described in the docstring of `Tree.mount()`:
>         script_name
>             ... 
>
>             It MUST NOT end in a slash. If the script_name refers to the
>             root of the URI, it MUST be an empty string (not "/").

**Review by webknjaz — CHANGES_REQUESTED** [review]

File: cherrypy/_cptree.py
Comment (webknjaz) [inline cherrypy/_cptree.py:224]: I'm not sure if it's a good idea for nested trees. Could you please check this too?

**Comment by webknjaz** [comment]
This change feels dangerous so I would also require a clear approval from @jaraco, not just myself.

File: cherrypy/test/test_core.py
Comment (jaraco) [reply cherrypy/test/test_core.py:839]: Normally, I'd be reluctant to ask someone to add tests for behavior outside the scope of the behavior they're trying to support, but in this case, I'm pretty sure there's a regression for untested behavior here, so adding those tests is important. If your `git` foo is good enough, consider adding the tests against master first, so you can see the expected passing behavior on the examples that webknjaz suggested, but also the failing behavior on `script_name='foo'`.

File: cherrypy/_cptree.py
Comment (jaraco) [inline cherrypy/_cptree.py:224]: I'm disinclined to continue to let the user supply an erroneous value and correct it for them. Why not instead hard-fail (raise an exception) when invalid values are input (similar to how `script_name=None` is handled)?

Comment (abtinmo) [reply cherrypy/_cptree.py:224]: fixes the script_name to prevent 404 error (not sure if it's necessary) , its what i said in my first comment. 
i don't think letting user to continue is a right solution. but in doc i saw this:
This should start with a slash

so i choose to fix error myself and throw a console error.
raise an exception is indeed a better solution

**Review by webknjaz — CHANGES_REQUESTED** [review]
AFAIR we discussed raising an exception for the invalid input. I'll wait for this change before attempting to review again.

**Comment by jaraco** [comment]
It's been a while since I last reviewed this change.

I remember many years ago trying to understand what is the purpose of this code and fixing bugs around it, so I'm uneasy adding more uncertainty to the implementation.

As I understand, we discussed and agreed to make the invalid input an error (Exception) instead of a warning, but there hasn't been any activity since, so I'm closing this as languishing. Feel free to revisit with a new PR.

**Pull request closed without merging**
